src/util/http/websocket/WebSocketSession.cpp

```diff
@@ -93,9 +93,11 @@ net::awaitable<void> WebSocketSession::handleSession(
     tcp::socket socket) {
   // Make sure access to new websocket is on a strand and therefore thread safe
   auto executor = co_await net::this_coro::executor;
-  AD_CONTRACT_CHECK(
-      executor.target<net::strand<net::io_context::executor_type>>());
-  co_await net::post(net::use_awaitable);
+  auto strandExecutor =
+      executor.target<net::strand<net::io_context::executor_type>>();
+  AD_CONTRACT_CHECK(strandExecutor);
+  AD_CONTRACT_CHECK(strandExecutor->running_in_this_thread());
+  AD_CONTRACT_CHECK(socket.get_executor() == *strandExecutor);
```

Comment:
`AD_CONTRACT_CHECK`s can and should be brought to fail in the test cases.
If this is not possible (e.g. because they are inside private functions and only can fail when there is a bug in the enclosing module) they should be `AD_CORRECTNESS_CHECK`s (the difference is in the way the coverage tool checks those).